File: 2028/solution.py
class Solution:
    def missingRolls(self, rolls: list[int], mean: int, n: int) -> list[int]:
        total = sum(rolls)
        count = n + len(rolls)
        missing = count * mean - total
        min_sum = n
        max_sum = n * 6
        if missing < min_sum or missing > max_sum:
            return []
        val = missing // n
        extra = missing % n
        res = [val + 1] * extra + [val] * (n - extra)
        return res

File: 2028/solution.go
package main

func missingRolls(m []int, mean int, n int) []int {
	mTotal := 0
	for _, v := range m {
		mTotal += v
	}
	nTotal := mean*(n+len(m)) - mTotal
	if nTotal < n || nTotal > n*6 {
		return []int{}
	}

	val := nTotal / n
	mod := nTotal % n
	res := make([]int, n)
	for i := range n {
		res[i] = val
		if i < mod {
			res[i]++
		}
	}
	return res
}